### DISCUSSION TITLE
How to improve the performance of ServiceLookUpEditor

### DISCUSSION THREAD

**Discussion by iamsachinupadhyay** [serenity-is/Serenity#6411] [Help]
We are trying to load UNLocation data stored in sql server into a service lookup editor? But there are multiple such fields in a dialogue. So it makes one network call for each editor reducing the performance. Is there a way that we fetch UNLocation data and make a ServiceLookUpEditor work in its best possible way that is to have pagination and other functionalities related to ServiceLookUpEditor?

**Comment by edwardch** [comment]
Does the lookup _have_ to be a ServiceLookupEditor? If not, you could try using a regular lookup that stores items in cache.
This will store all data for the dropdown in the user's cache, so if there is _lots_ of data your UX of initial load may be affected.

If you require it to be a ServiceLookupEditor, you should check what query is hitting your SQL server, then ensuring there is adequate indexing to make sure your service lookups are as fast as they can be. And also check the properties on ServiceLookupEditor to keep the query and results light. Check parameters like PageSize, Sort, etc...

**Reply by iamsachinupadhyay** [reply]
Thanks for the quick response.

We are using ServiceLookupEditor only and SQL indexes are there on the table.

Retrieve request accept entity id. Is there any way where I can have a list of entity ids in the retrieve request?

**Reply by edwardch** [reply]
You could try put a FilterField / FilterValue, or CascadeField / CascadeValue on the lookup property and pass in your Entity Id?

You can actually set this in your ClassDialog.ts, I think it would be something like this:

```ts
this.form.ForeignKeyFieldID.filterField = 'SomeField';
this.form.ForeignKeyFieldID.filterValue = Q.toId(someValue);
```

**Reply by iamsachinupadhyay** [reply]
We have two fields destination and origin and both have the data from the same unloc table.

**Reply by iamsachinupadhyay** [reply]
[edwardch] Any suggestions on the this query? We have two fields destination and origin and both have the data from the same unloc table. How we can have one call only so that we can pre-populate two fields

**Reply by edwardch** [reply]
Not sure what else to suggest here, but I suppose worth finding out a couple of things...

- How "poor" is the performance?
- How many times is this table accessed per minute (50 times, 500 times, 50000 times)? 
- How big is this location table? 
- How big/small is your database server?
- How far/close is the database to the app?
- etc
